src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -993,12 +1007,41 @@ static ImmutableArray<Symbol> getAllTypeAndRequiredMembers(TypeSymbol containing
                             foreach (var (_, requiredMember) in requiredMembers)
                             {
                                 // We want to assume that all required members were _not_ set by the chained constructor
-                                builder.Add(getFieldSymbolToBeInitialized(requiredMember));
+                                builder.AddRange(getAllMembersToBeDefaulted(requiredMember));
                             }
 
                             return builder.ToImmutableAndFree();
                         }
 
+                        static ImmutableArray<Symbol> getAllMembersToBeDefaulted(Symbol requiredMember)
+                        {
+                            Debug.Assert(requiredMember.IsRequired());
+
+                            if (requiredMember is FieldSymbol)
+                            {
+                                return ImmutableArray.Create<Symbol>(getFieldSymbolToBeInitialized(requiredMember));
+                            }
+                            else
+                            {
+                                var property = (PropertySymbol)requiredMember;
+                                // If the set method is null (ie missing), that's an error, but we'll recover as best we can
+                                var setMethodNotNullMembers = property.SetMethod?.NotNullMembers ?? property.NotNullMembers;
+                                var builder = ArrayBuilder<Symbol>.GetInstance(1 + setMethodNotNullMembers.Length);
+
+                                builder.Add(getFieldSymbolToBeInitialized(property));
+
+                                foreach (var notNullMemberName in setMethodNotNullMembers)
+                                {
+                                    foreach (var member in property.ContainingType.GetMembers(notNullMemberName))
+                                    {
+                                        builder.Add(getFieldSymbolToBeInitialized(member));
+                                    }
+                                }
+
+                                return builder.ToImmutableAndFree();
```

Comment:
>return builder.ToImmutableAndFree(); [](http://example.com/codeflow?start=32&length=36)

Do we really need to allocate an immutable array here? It looks like we could handle a regular IEnumerable at the call sites, and this method could be an iterator.